apps/game_backend/lib/game_backend/curse_of_mirra/map_configuration.ex

```diff
@@ -71,6 +74,52 @@ defmodule GameBackend.CurseOfMirra.MapConfiguration do
       |> cast_embed(:position)
       |> cast_embed(:vertices)
       |> validate_required([:name, :position, :radius, :shape, :type])
+      |> MapConfiguration.validate_shape()
+    end
+  end
+
+  defmodule Pool do
+    @moduledoc """
+    Pool embedded schema to be used by MapConfiguration
+    """
+    alias GameBackend.CurseOfMirra.MapConfiguration
+    use GameBackend.Schema
+
+    @derive {Jason.Encoder, only: [:name, :position, :radius, :shape, :vertices, :effects_to_apply]}
+    embedded_schema do
+      field(:name, :string)
+      field(:radius, :decimal)
+      field(:shape, Ecto.Enum, values: [:circle, :polygon])
+      field(:effects_to_apply, {:array, :string})
```

Comment:
This might work right now, but once we have the effects in the database, this will no longer work 😬

Reply:
Hmmm maybe we should takle this when we implement the effects into the database, i think there's no easy solution applicable on this pr 🤔

Reply:
Yeah, no problem, we will not address this here